### PR TITLE
[scala-package] add apache license header and corresponding scalastyle config entry

### PR DESCRIPTION
**Pull request by CodingCat**
IIRC, this is a must-have after being incubator project......

@javelinjs @terrytangyuan......

### CODE REVIEW

**Comment by piiswrong** [comment]
does it have to be in every file?

**Comment by CodingCat** [comment]
I remember Spark community once had the same conversation....the final result is yes....

**Comment by CodingCat** [comment]
@piiswrong search `Each source file should include the following license header` https://www.apache.org/legal/src-headers.html

File: scala-package/init-native/linux-x86_64/pom.xml [hunk 1]
```diff
@@ -10,7 +10,6 @@
     <relativePath>../pom.xml</relativePath>
   </parent>
 
-  <groupId>ml.dmlc.mxnet</groupId>
```
Comment (piiswrong) [inline scala-package/init-native/linux-x86_64/pom.xml]: why remove?

Comment (CodingCat) [inline scala-package/init-native/linux-x86_64/pom.xml]: this will inherit from the parent

**Comment by CodingCat** [comment]
thanks

**Pull request merged**